cli/throttle.go

```diff
@@ -0,0 +1,33 @@
+package cli
+
+import (
+	"sync"
+	"time"
+)
+
+type Throttle struct {
+	rate     int // requests per second
+	requests int // total requests
+}
+
+// simple rate limitter
+func (l Throttle) RateLimit(f func()) {
+	executed := 0
+
+	ticker := time.Tick(time.Second)
+	var wg = &sync.WaitGroup{}
+	wg.Add(l.requests)
+	for ; true; <-ticker {
+		if executed >= l.requests {
+			break
+		}
+		for i := 0; i < l.rate; i++ {
+			go func() {
+				f()
+				executed++
+				wg.Done()
+			}()
+		}
+	}
+	wg.Wait()
```

Comment:
Some questions/concerns:

1. What happens if a request takes more than a second? Seems like we'd get many in flight at the same time.
2. If a request finishes really fast there is no opportunity to create new ones until the full second has passed, even if a sliding time window would allow a new request.
3. This assumes all the requests are the same, but #134 suggests they might get generated for each API operation. How will that work?